docs/syntax_and_semantics/platform_support.md

```diff
@@ -14,41 +14,56 @@ Tier 1 platforms can be thought of as “guaranteed to work”. Specifically the
-| arm-linux-gnueabihf | ✓ | ✓ | ARM 32-bit Linux (GNU libc, hardfloat) |
-| i386-linux-gnu | ✓ | ✓ | 32-bit Linux (kernel 2.6.18+, GNU libc) |
-| i386-linux-musl | ✓ | ✓ | 32-bit Linux (MUSL libc) |
-| x86_64-linux-musl | ✓ | ✓ | 64-bit Linux (MUSL libc) |
-| x86_64-openbsd | ✓ | ✓ | 64-bit OpenBSD (6.x) |
-| x86_64-freebsd | ✓ | ✓ | 64-bit FreeBSD (12.x) |
+| Target | Description | Supported versions | Comment |
+| ------ | ----------- | ------------------ | ------- |
+| `aarch64-darwin` | Aarch64 macOS (Apple Silicon) | 11+ | ❌ tests ✅ builds
+| `aarch64-linux-gnu` | Aarch64 Linux (hardfloat) | GNU libc 2.26+ | ✅ tests ❌ builds
+| `aarch64-linux-musl` | Aarch64 Linux (hardfloat) | MUSL libc 2.26+ | ✅ tests ❌ builds
+| `arm-linux-gnueabihf` | Aarch32 Linux (hardfloat) | GNU libc 2.26+ | ❌ tests ❌ builds
+| `i386-linux-gnu` | x86 Linux | kernel 4.14+, GNU libc 2.26+<br> *(expected to work on kernel 2.6.18+)* | ❌ tests ❌ builds
+| `i386-linux-musl` | x86 Linux | kernel 4.14+, MUSL libc 1.2+<br> *(expected to work on kernel 2.6.18+)* | ❌ tests ❌ builds
+| `x86_64-linux-musl` | x64 Linux | kernel 4.14+, MUSL libc 1.2+<br> *(expected to work on kernel 2.6.18+)* | ✅ tests ✅ builds
+| `x86_64-openbsd` | x64 OpenBSD | 6+ | ❌ tests ❌ builds
+| `x86_64-freebsd` | x64 FreeBSD | 12+ | ❌ tests ❌ builds
 
 ***
 
 ## Tier 3
 
-Tier 3 platforms are those which the Crystal codebase has some sort of support for, but which are not built or tested automatically, and may not work. Official builds are not available.
+Tier 3 platforms can be though of as “partially works”.
+
+The Crystal codebase has support for these platforms, but there are some major limitations.
+Most typically, some parts of the standard library are not supported completely.
+
+| Target | Description | Supported versions | Comment |
+| ------ | ----------- | ------------------ | ------- |
+| `x86_64-windows-msvc` | x64 Windows (MSVC ) | 7+ | 🟡 tests<br> ✅ builds |
+| `aarch64-linux-android` | aarch64 Android  | Bionic C runtime, API level 28+ | ❌ tests<br> ❌ builds |
+| `x86_64-unknown-dragonfly` | x64 DragonFlyBSD | | ❌ tests<br> ❌ builds |
+| `x86_64-unknown-netbsd` | x64 NetBSD | | ❌ tests<br> ❌ builds |
+| `wasm32-unknown-wasi` | WebAssembly (WASI libc) | Wasmtime 2+ | 🟡 tests |
+| `interpreted` | Crystal's interpreter (`crystal i`) | | 🟡 tests |
 
-| Target | Compiler | Std | Description |
-| ------ | -------- | --- | ----------- |
-| x86_64-windows-msvc |  |  | 64-bit MSVC (Windows 7+) |
-| aarch64-linux-android |  |  | ARM 64-bit Android (Bionic C runtime, API level 28+) |
-| x86_64-unknown-dragonfly | | | 64-bit DragonFlyBSD |
-| x86_64-unknown-netbsd | | | 64-bit NetBSD |
-| wasm32-unknown-wasi | | | WebAssembly (WASI libc) |
+!!! info "Legend"
+    * ❌ means automated tests or builds are not available
+    * ✅ means automated tests or builds are available
+    * 🟡 means automated test are available, but the implementation is incomplete
```

Comment:
Two stylistic suggestions: b/w icons and avoid the red cross as it suggests tests are failing, which is a stronger statement.

```suggestion
    * ◻︎ means automated tests or builds are not available
    * ☑︎ means automated tests or builds are available
    * ● means automated test are available, but the implementation is incomplete
```

Reply:
PS I don't know how these icons look in the page (but I know I don't like the current ones)

Reply:
IMO the visuals fit very nicely with the rest of the page. Your suggested icons do not. I'll see if I can find some alternative suggestions.

Reply:
> the visuals fit very nicely with the rest of the page.

Not critical, but I disagree: the 3d looking icons do not match anything on that flat-looking page. Anyway, the most critical aspect to me is avoiding the red cross.

Reply:
For reference, I'm seeing this:

![grafik](https://user-images.githubusercontent.com/466378/228279507-9cf18d1e-2198-453a-8b1c-fdc4c621cad7.png)

Reply:
The icon emojis unfortunately are quite a bit of a mess 😞
Icon size and position is inconsistent, even for the red and yellow circle which are from the same icon set:
![grafik](https://user-images.githubusercontent.com/466378/228279943-c55e9e9d-cdce-4a2d-9718-b8f27d83258b.png)

Reply:
Here are some suggestions for icon combinations:

![grafik](https://user-images.githubusercontent.com/466378/228285458-5efc3022-9206-4214-b9a9-54bec3913dc6.png)

Reply:
I prefer coloured indicators for easy readability. I like the Unicode icons best for this, but unfortunately, the visuals are platform-specific and thus inconsistent. I could not recreate a decent coloured set with mkdocs icons (Temoji), they're inconsistent (the red circle is too big, the others are sitting too low).
Of these other alternatives, I think Material circles and Material checkbox circles are looking best.

Another option would be to use coloured icons via different means (for example directly embedded images or SVGs). This would make the source a bit nasty but IMO has the best results.

Reply:
For reference, this is what I see (c'mon Mac!)
<img width="645" alt="image" src="https://user-images.githubusercontent.com/5183634/228288904-d117417e-e5e3-42bd-8a9b-6c8d5c89ba64.png">

Reply:
Yeah, it looked good but didn't work out well in practice. In the tables it looked too much like checkboxes and was asking to be clicked 😆 

So I came up with a small modification, and I'm quite happy with it. The empty circle has a dashed line to symbolize absence and the full circle has a check mark in it to symbolize success. Also I changed the fill ratio of the partial circle to 5/8 which looks more pleasant.
![grafik](https://user-images.githubusercontent.com/466378/228343235-624c8921-47e0-4b53-af95-651cf6553795.png)